backend/src/config/serverConfig.ts

```diff
@@ -39,13 +43,18 @@ class Server {
   }
 
   private routes() {
-    this.app.use(teamRoutes); //recommend this.app.use(this.paths.teams, teamRoutes)
-    this.app.use(memberRoutes); //recommend this.app.use(this.paths.member, memberRoutes)
+    this.app.use(this.paths.teams, teamRoutes);
+    this.app.use(this.paths.member, memberRoutes);
     this.app.use(this.paths.user, userRoutes);
     this.app.use(this.paths.auth, authRoutes);
   }
 
   public listen() {
+    this.app.get("/api-docs", (req, res) => {
+      res.sendFile(path.join(__dirname, "../utils/backend_specs.html"));
+    });
+    generateDocs();
+
```

Comment:
**Single Responsibility Principle** applies to methods and variables as well. The `listen` method is for starting up the server not defining paths and registering request handlers. 

Your path, `api-docs` belongs in the `paths` object in the constructor, `generateDocs()` belongs in `init` after the routes have been registered and your route definition `this.app.get...` belongs in the `routes` method and it shouldn't use a string literal but rather a member of the `paths` object. 

There is an implied interface in the class' structure so maintaining the conventions makes sure you don't violate **Liskov Substitution Principle** as well.

Reply:
Thanks for calling that out!
I understand now and have implemented the same in the following commit. 
I didn't create a router file and controller as it just feels like overkill for a single route definition which is never going to have more than one. Does that make sense?

Reply:
It's okay you can just provide the inline definition like you did, no need for separate files

Reply:
Alright!